evalml/pipelines/components/estimators/regressors/timeseries_baseline_regressor.py

```diff
@@ -0,0 +1,73 @@
+import numpy as np
+import pandas as pd
+
+from evalml.model_family import ModelFamily
+from evalml.pipelines.components.estimators import Estimator
+from evalml.problem_types import ProblemTypes
+from evalml.utils.gen_utils import (
+    _convert_to_woodwork_structure,
+    _convert_woodwork_types_wrapper
+)
+
+
+class TimeSeriesBaselineRegressor(Estimator):
+    """Time series regressor that predicts using the naive forecasting approach.
+
+    This is useful as a simple baseline regressor for time series problems
+    """
+    name = "Time Series Baseline Regressor"
+    hyperparameter_ranges = {}
+    model_family = ModelFamily.BASELINE
+    supported_problem_types = [ProblemTypes.TIME_SERIES_REGRESSION]
+
+    def __init__(self, gap=1, random_state=0, **kwargs):
```

Comment:
Also, I'm assuming it doesn't make sense if gap is negative (or 0?)? Can we add a check for that and raise / warn if that's true?